packages/vaadin-grid-pro/src/vaadin-grid-pro-inline-editing-mixin.js

```diff
@@ -102,11 +102,8 @@ export const InlineEditingMixin = (superClass) =>
       // Prevent click/selection on edit column
       this.$.table.addEventListener('click', (e) => {
         const column = this.getEventContext(e).column;
-        if (column && this._isEditColumn(column)) {
-          e.preventDefault();
```

Comment:
This `preventDefault()` is most probably not required here since its removal doesn't break any tests (neither ITs nor unit tests) while its presence here creates a problem for the checkbox. The new checkbox's implementation relies on the `change` event fired on the slotted input, but the `change` event doesn't get fired if the previous `click` event has been prevented.

Reply:
It was agreed that @yuriy-fix will spend some extra time looking into reasons for that `preventDefault()` and, if there is no anything unaccounted for, we will go without `preventDefault()`.

Reply:
I have tested with the GridPro Flow and found out that it was preventing selection:

![not-prevented](https://user-images.githubusercontent.com/29372419/134139412-42fcda01-331a-41b7-b80a-d611884583d6.gif)
![prevented](https://user-images.githubusercontent.com/29372419/134139441-cfb50ef7-1f8e-4116-aa90-eeaa49c4a459.gif)

Reply:
@vursen I think you need to add an explicit check to detect whether `event.composedPath()` includes a checkbox. In that case the click event should not be prevented. Also the tests need to be added to cover these cases.

Reply:
While fixing the `preventDefault()` issue, I found another similar issue: the checkbox doesn't respond in the `editOnClick` mode so I fixed both.